test/libcommonserver/io/testcheckifpathexists.cpp

```diff
@@ -378,106 +411,67 @@ void TestIo::testCheckIfPathExistsWithSameNodeIdSimpleCases() {
-        const SyncPath subdir = temporaryDirectory.path() / "permission_less_subdirectory";
-        std::filesystem::create_directory(subdir);
-
-        const SyncPath targetPath = _localTestDirPath / "test_pictures/picture-1.jpg";
-        const SyncPath path = subdir / "regular_file_symbolic_link";
-        std::filesystem::create_symlink(targetPath, path);
+        const LocalTemporaryDirectory temporaryDirectory("TestIo");
+        const SyncPath targetPath = _localTestDirPath / "non_existing_dir";  // It doesn't exist.
+        const SyncPath path = temporaryDirectory.path() / "dir_junction";
 
-        _testObj->setReadSymlinkFunction([&subdir](const SyncPath &path, std::error_code &ec) -> SyncPath {
-            std::filesystem::permissions(subdir, std::filesystem::perms::owner_exec, std::filesystem::perm_options::remove);
-            return std::filesystem::read_symlink(path, ec);
-        });
+        IoError ioError = IoErrorUnknown;
+        CPPUNIT_ASSERT(_testObj->createJunctionFromPath(targetPath, path, ioError));
+        CPPUNIT_ASSERT(ioError == IoErrorSuccess);
 
         bool existsWithSameId = false;
         NodeId otherNodeId;
-        IoError ioError = IoErrorUnknown;
         NodeId nodeId;
 
         CPPUNIT_ASSERT(_testObj->getNodeId(path, nodeId));
         CPPUNIT_ASSERT(_testObj->checkIfPathExistsWithSameNodeId(path, nodeId, existsWithSameId, otherNodeId, ioError));
-
-        // Restore permission to allow subdir removal
-        std::filesystem::permissions(subdir, std::filesystem::perms::owner_exec, std::filesystem::perm_options::add);
-#ifdef _WIN32
         CPPUNIT_ASSERT(existsWithSameId);
-#else
-        CPPUNIT_ASSERT(!existsWithSameId);  // getLocalNodeId returns an empty string because of the denied access.
-        CPPUNIT_ASSERT(ioError == IoErrorAccessDenied);
-#endif
-
-        _testObj->resetFunctions();
+        CPPUNIT_ASSERT(ioError == IoErrorSuccess);
     }
 
-    // Reading a regular symbolic link that is removed right after `filesystem::is_simlink` was called.
+    // A Windows junction on a regular target file.
     {
-        const SyncPath targetPath = _localTestDirPath / "test_pictures";
-        const LocalTemporaryDirectory temporaryDirectory;
-        const SyncPath path = temporaryDirectory.path() / "regular_dir_symbolic_link";
-        std::filesystem::create_symlink(targetPath, path);
+        const LocalTemporaryDirectory temporaryDirectory("TestIo");
+        const SyncPath targetPath = _localTestDirPath / "test_pictures/picture-1.jpg";
+        const SyncPath path = temporaryDirectory.path() / "dir_junction";
 
-        _testObj->setReadSymlinkFunction([](const SyncPath &path, std::error_code &ec) -> SyncPath {
-            std::filesystem::remove(path);
-            return std::filesystem::read_symlink(path, ec);
-        });
+        IoError ioError = IoErrorUnknown;
+        CPPUNIT_ASSERT(_testObj->createJunctionFromPath(targetPath, path, ioError));
```

Comment:
It's surprising that it works. Good to know anyway.

Reply:
Not sure about what you are talking about here?

(I just change the name of the junction if it was what you where talking about 👍🏾 @herve-er
[More explicit name for test junction.](https://github.com/Infomaniak/desktop-kDrive/pull/239/commits/a3f6111fcb7cd151fd00d1f7e80cbc9b9709b437))

Reply:
I was referring to the outcome of the test: we can create successfully a junction whose target is a file, as opposed to a directory. It surprised me when I re-discovered it.